1.9/administration/upgrading.md

```diff
@@ -18,153 +18,75 @@ This document provides instructions for upgrading a DC/OS cluster from version 1
+- For Mesos compatibility reasons, we recommend upgrading any running Marathon-on-Marathon instances on DC/OS 1.8 to Marathon version 1.3.5 before upgrading to DC/OS 1.9.
+- You must have access to copies of the config files used with DC/OS 1.8: `config.yaml` and `ip-detect`.
 - You must be using systemd 218 or newer to maintain task state.
 - All hosts (masters and agents) must be able to communicate with all other hosts on all ports, for both TCP and UDP.
 - In CentOS or RedHat, install IP sets with this command (used in some IP detect scripts): `$ sudo yum install -y ipset`
 - You must be familiar with using `systemctl` and `journalctl` command line tools to review and monitor service status. Troubleshooting notes can be found at the end of this [document](#troubleshooting).
 - You must be familiar with the [Advanced DC/OS Installation Guide][advanced-install].
 - You should take a snapshot of ZooKeeper prior to upgrading. Marathon supports rollbacks, but does not support downgrades.
+- The full DC/OS version string that you are upgrading from.
+  - In 1.8 this can be found in the lower left corner of the DC/OS UI when screen is maximized.
+  - In 1.9 this can be found under the Cluster menu.
+
+## Supported Upgrade Paths
+
+- 1.8.latest => 1.9.latest
+- 1.9.x => 1.9.(x+1)
+- 1.9.x => 1.9.x (for configuration changes)
 
 ## Instructions
 
 ### Bootstrap Nodes
 
-1.  Copy and update the DC/OS 1.7 `config.yaml` and `ip-detect` files to a new, clean folder on your bootstrap node.
+1.  Copy and update the DC/OS 1.8 `config.yaml` and `ip-detect` files to a new, clean folder on your bootstrap node.
 
     **Important:**
 
     *  You cannot change the `exhibitor_zk_backend` setting during an upgrade.
-    *  The syntax of the DC/OS 1.8 `config.yaml` differs from that of DC/OS 1.7. For a detailed description of the 1.8 `config.yaml` syntax and parameters, see the [documentation](/docs/1.9/administration/installing/custom/configuration-parameters/).
+    *  The syntax of the DC/OS 1.9 `config.yaml` differs from that of DC/OS 1.8. For a detailed description of the 1.9 `config.yaml` syntax and parameters, see the [documentation](/docs/1.9/administration/installing/custom/configuration-parameters/).
 
-1.  After you have merged your 1.7 `config.yaml` into the 1.8 `config.yaml` format, you can build your installer package:
+1.  After you have converted your 1.8 `config.yaml` into the 1.9 `config.yaml` format, you can build your installer package:
 
     1.  Download the file `dcos_generate_config.sh`.
     1.  Generate the installation files.
-    1.  Disable Docker restarts in `dcos_install.sh` by using this command:
-
         ```
-        sed -i -e "s/systemctl restart systemd-journald//g" -e "s/systemctl restart docker//g" genconf/serve/dcos_install.sh
+        dcos_generate_config.sh --generate-node-upgrade-script <installed_cluster_version_upgrading_from>
```

Comment:
The installer calls this arg `installed_cluster_version`, so let's use that here as well. https://github.com/dcos/dcos/blob/18c76a2b4b24aab0c4107bae9c7191a68e6de174/dcos_installer/cli.py#L271